geode-core/src/test/java/org/apache/geode/distributed/internal/MaxLongGaugeTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.distributed.internal;
+
+import static org.apache.geode.internal.statistics.StatisticDescriptorImpl.createLongGauge;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.geode.StatisticDescriptor;
+import org.apache.geode.internal.statistics.StatisticsTypeImpl;
+import org.apache.geode.internal.statistics.StripedStatisticsImpl;
+
+public class MaxLongGaugeTest {
+  private MaxLongGauge maxLongGauge;
+  private StripedStatisticsImpl fakeStatistics;
+
+  @Before
+  public void setup() {
+    StatisticDescriptor descriptor =
+        createLongGauge("1", "", "", true);
+    StatisticDescriptor descriptor2 =
+        createLongGauge("2", "", "", true);
+
+    StatisticDescriptor[] descriptors = {descriptor, descriptor2};
+    StatisticsTypeImpl statisticsType = new StatisticsTypeImpl("abc", "test",
+        descriptors);
+    fakeStatistics = new StripedStatisticsImpl(
+        statisticsType,
+        "def", 12, 10,
+        null);
+  }
+
+  @Test
+  public void recordMax_singleRecord() {
+    MaxLongGauge maxLongGauge = new MaxLongGauge(0, fakeStatistics);
```

Comment:
how about reading the id from the statisticsType instead of using this hard coded 0? That way if the id impl changes again this test will still work.